### PR TITLE
amam/style_update

### PR DESCRIPTION
**Pull request by cakasuma**
1 As per webtrader historical data:
Fix dropdown
Datepicker & timepicker
Window menu

2 Fixed height
Remove fixed height from password window explanations --> fixes text overlap

### CODE REVIEW

File: src/historical-data/historical-data.es6 [hunk 1]
```diff
@@ -20,8 +20,8 @@ const buildMenu = ($root, instrumentName, callback) => {
       _.delay(() => buildMenu($root, instrumentName, callback), 1000);
       return;
    }
-   const menu = `<ul>${
-      markets.map(m => `<li><div>${m.display_name}</div><ul>${
+   const menu = `<ul >${
```
Comment (OskarAhl) [inline src/historical-data/historical-data.es6]: remove the empty space e.g. <ul>

File: src/historical-data/historical-data.es6 [hunk 2]
```diff
@@ -20,8 +20,8 @@ const buildMenu = ($root, instrumentName, callback) => {
...
+      markets.map(m => `<li ><div>${m.display_name}</div><ul>${
```
Comment (OskarAhl) [inline src/historical-data/historical-data.es6]: same here 😄

File: src/historical-data/historical-data.es6 [hunk 3]
```diff
@@ -54,26 +54,32 @@ const buildDatetime = (dialog, $root, callback) => {
...
          .timepicker({
-            showCloseButton : true,
-            beforeShow: (input, inst) => inst.tpDiv.css({marginLeft: '-120px'}),
+            showCloseButton : false,
+            beforeShow: (input, inst) => inst.tpDiv.css(
+            {
+                marginLeft: '-120px',
+                marginTop: '6px'
+            }
+
```
Comment (OskarAhl) [inline src/historical-data/historical-data.es6]: Usually we follow the format of `({
...
})`

**Comment by cakasuma** [comment]
Alright, will do

File: src/historical-data/historical-data.scss [hunk 4]
```diff
@@ -3,13 +3,13 @@
       padding : 0 !important;
       position: relative;
    }
-   .instrumnet-dropdown {
+   .instrument-dropdown {
       position: absolute;
-      background: linear-gradient(#2A3052,#15212D);
-      height: 34px;
+
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: You can remove this empty space

File: src/historical-data/historical-data.scss [hunk 5]
```diff
@@ -34,45 +35,95 @@
...
    .date-time {
+    
       position: absolute;
-      background: linear-gradient(#2A3052,#15212D);
-      height: 34px;
-      width: 210px;
-      left: 340px;
-      top: 0;
+      
+
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: These empy spaces too 😄

File: src/historical-data/historical-data.scss [hunk 6]
```diff
@@ -34,45 +35,95 @@
...
+        background-repeat: no-repeat;
+        margin: 0;
+        white-space: nowrap;
       }
+
    }
    .ui-dialog-titlebar {
       .ui-dialog-title {
-         font-size: 16px;
+
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: This empty space also

File: src/historical-data/historical-data.scss [hunk 7]
```diff
@@ -34,45 +35,95 @@
...
+    border-right: 3px solid #f2f2f2;
+}
+
+.ui-widget-content button {
+    float: right !important;
+}
+
+.ui-datepicker-close {
+    display: none;
+}
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: Perhaps we could move all these to be "under" .historical-data-dialog so they don't conflict with css in other windows

File: src/historical-data/historical-data.scss [hunk 8]
```diff
@@ -34,45 +35,95 @@
...
+
+.ui-timepicker td {
+    border: none;
+
+}
+
+.ui-timepicker td a {
+    display: block;
+    padding: .2em .3em;
+    width: 1.2em;
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: Why are we using `em`s here?

File: src/main.css [hunk 9]
```diff
@@ -371,6 +371,42 @@ button[disabled] {
   background: none;
   border: none;
 }
+.ui-datepicker-close {
+	display: none;
+}
+.ui-widget-content button {
+	float: right !important;
```
[the remark this answers is not in the document]
Comment (cakasuma) [reply src/main.css]: this is actually used for tradingtimes to remove the done button and to make the today button go into right side.

File: src/historical-data/historical-data.scss [hunk 10]
```diff
@@ -1,78 +1,77 @@
+$primary-color: #2a3052;
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: For colors we follow `$COLOR_DESCRIPTION` e.g. `$COLOR_DARK_BLUE`

File: src/historical-data/historical-data.es6 [hunk 11]
```diff
@@ -34,16 +34,16 @@ const buildMenu = ($root, instrumentName, callback) => {
...
    $menu.menu();
 };
 
 const buildDatetime = (dialog, $root, callback) => {
     const datetime = $(`<div class='date-time'>
-                    <input type="button" class="date" tab-index="-1"></input>
-                    <input type="button" class="time" value="00:00"></input>
+                    <input type="text" class="date" tab-index="-1" readonly></input>
+                    <input type="text" class="time" tab-index="-1" value="00:00" readonly></input>
           </div>`);
```
Comment (OskarAhl) [inline src/historical-data/historical-data.es6]: 👍

File: src/historical-data/historical-data.scss [hunk 12]
```diff
@@ -1,78 +1,78 @@
...
+		.title {
+			background-image: url('../images/thin-arrowheads-pointing-down.svg') !important;
+			padding-left: 7px;
+			line-height: 34px;
+			font-size: 18px;
+			margin: auto;
+			background-size: 10px 10px !important;
+			background-position: 90px 50% !important;
+			background-repeat: no-repeat;
+			margin: 0;
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: 2x margins (I think we can remove one)

File: src/historical-data/historical-data.scss [hunk 13]
```diff
@@ -74,4 +73,32 @@ $COLOR_WHITE: #ffffff;
 			border: 0.2px solid $COLOR_DARK_BLUE;
 		}
 	}
+}
+
+a.ui-state-default.ui-state-active {
+    border: 2px solid #2a3052!important;
+    background: #2a3052!important;
+    color: #fff!important;
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: Can you add the colors as variables and define them on top?

File: src/historical-data/historical-data.es6 [hunk 14]
```diff
@@ -68,7 +68,7 @@ const buildDatetime = (dialog, $root, callback) => {
              maxDate : moment.utc().toDate(),
              currentText: 'Today'.i18n()
          })
-         .datepicker("setDate", moment.utc().subtract(1, "years").toDate().toISOString().slice(0, 10));
+         .datepicker("setDate", "0");
```
Comment (OskarAhl) [inline src/historical-data/historical-data.es6]: Let's use single quotation marks

File: src/historical-data/historical-data.scss [hunk 15]
```diff
@@ -74,4 +73,32 @@ $COLOR_WHITE: #ffffff;
...
+    background-image: url(data:image/svg+xml;charset=utf8;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmcnIHZpZXdCb3g9Jy0yMzM5OC4yOTUgLTIxMTE4Ljk0MSAxNC44MiAyNCc+PHBhdGggZmlsbD0nd2hpdGUnIGQ9J004LjU5LDI3LjE4LDE3Ljc1LDE4LDguNTksOC44MiwxMS40MSw2bDEyLDEyLTEyLDEyWicgdHJhbnNmb3JtPSd0cmFuc2xhdGUoLTIzMzc0Ljg4NSAtMjEwODguOTQxKSByb3RhdGUoMTgwKScvPjwvc3ZnPg==) !important;
+    background-position: initial;
+}
+
+.ui-icon-circle-triangle-w:after {
+    content: unset;
+}
+
+.ui-icon-circle-triangle-e:after {
+    content: unset;
```
Comment (OskarAhl) [inline src/historical-data/historical-data.scss]: As dicussed we can add this to `main.css` so all datepickers have the same style 👍

File: src/main.css [hunk 16]
```diff
@@ -371,6 +388,116 @@ button[disabled] {
...
+    background-position: initial;
+}
+
+.ui-icon-circle-triangle-w:after {
+    content: unset;
+}
+
+.ui-icon-circle-triangle-e:after {
+    content: unset;
+}
```
Comment (OskarAhl) [inline src/main.css]: The issue with overriding the jQuery styles in main is that it's a lot of effort to maintain (every time binary-style changes we need to update this). 

We should look for a way to migrate away from jQuery UI? (but that's out of the scope for this card 😄 )

**Pull request merged**